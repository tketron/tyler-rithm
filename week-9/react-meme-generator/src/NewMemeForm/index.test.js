import React from 'react'; // for JSX
import { shallow } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import NewMemeForm from '.'; // import the component itself

describe('<NewMemeForm />', () => {
  let wrapper;
  it('renders', () => {
    // smoke test!!!
    wrapper = shallow(<NewMemeForm addMeme={jest.fn()} />);
  });
  it('matches snapshot', () => {
    //snapshot test
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
