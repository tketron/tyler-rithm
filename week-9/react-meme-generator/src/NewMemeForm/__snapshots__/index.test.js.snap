// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMemeForm /> matches snapshot 1`] = `
<div>
  <h2>
    Make a New Meme
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="url"
    >
      Image URL
    </label>
    <input
      id="form_url"
      name="url"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="topText"
    >
      Top Text
    </label>
    <input
      id="form_topText"
      name="topText"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="bottomText"
    >
      Bottom Text
    </label>
    <input
      id="form_bottomText"
      name="bottomText"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      id="form_submit"
      type="submit"
    >
      Generate Meme!
    </button>
  </form>
</div>
`;
